caddyhttp/proxy/upstream.go

```diff
@@ -24,6 +25,8 @@ type staticUpstream struct {
 	from              string
 	upstreamHeaders   http.Header
 	downstreamHeaders http.Header
+	stop              chan struct{}
+	wg                sync.WaitGroup
```

Comment:
Could be worth adding a comment what these two fields do: stop signals to stop any running goroutines, and wg is used to wait on those goroutines to finish stopping.